### PR TITLE
JIT: ARM64 - Enable value-numbering for TYP_MASK

### PR DESCRIPTION
**Pull request by TIHan**

### CODE REVIEW

**Comment by kunalspathak** [comment]
I assume this is based on https://github.com/dotnet/runtime/pull/99743?

File: src/coreclr/jit/valuenum.cpp [hunk 1]
```diff
@@ -10934,7 +10929,8 @@ void Compiler::fgValueNumberTreeConst(GenTree* tree)
             tree->gtVNPair.SetBoth(vnStore->VNForSimd64Con(simd64Val));
             break;
         }
-
+#endif // TARGET_XARCH
+#if defined(TARGET_XARCH) || defined(TARGET_ARM64)
```
Comment (kunalspathak) [inline src/coreclr/jit/valuenum.cpp:10933]: fix the comment for `#endif`

**Comment by kunalspathak** [comment]
Looks like @mikabl-arm also have it in https://github.com/dotnet/runtime/pull/103818.

**Comment by TIHan** [comment]
Closing in favor of https://github.com/mikabl-arm/dotnet-runtime/tree/TYP_MASK-ARM64

**Pull request closed without merging**